x-pack/plugins/security/public/components/confirm_modal.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FunctionComponent } from 'react';
+import {
+  EuiButton,
+  EuiButtonProps,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalProps,
+  EuiOverlayMask,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+export interface ConfirmModalProps extends Omit<EuiModalProps, 'onClose' | 'initialFocus'> {
+  confirmButtonText: string;
+  confirmButtonColor?: EuiButtonProps['color'];
+  isLoading?: EuiButtonProps['isLoading'];
+  isDisabled?: EuiButtonProps['isDisabled'];
+  onCancel(): void;
+  onConfirm(): void;
+  ownFocus?: boolean;
+}
+
+/**
+ * Component that renders a confirmation modal similar to `EuiConfirmModal`, except that
+ * it adds `isLoading` prop, which renders a loading spinner and disables action buttons,
+ * and `ownFocus` prop to render overlay mask.
```

Comment:
This would make a great EUI issue 😉 . It'd be great to continue to use the actual EuiConfirmModal and just flesh it out with these feature requests. That way the "official" EUI one and the Kibana one don't get out of sync.

Reply:
Completely agree and I'm happy to open an issue but I wanted to establish what good pattern are within the scope of our project first, validate them against a few more use cases and then potentially file issues / back-port to EUI. You are correct though, this one is fairly obvious - We need a way to display loading states in confirmation modals ;)

I've opened an issue here: https://github.com/elastic/eui/issues/4411

I'm glad you mentioned it though. My plan was to eventually propose moving some of these components and hooks into a shared module in Kibana or even EUI once they're a bit more mature. I think EUI would be a great place to share, not just UI components, but UX behaviours as well. Forms is a good example for this - Every app, sometimes even different pages within the same app, have different behaviours for when inline validation get triggers, when submit buttons get disabled and how loading states are displayed within the form. That's a direct result of there not being an "EUI way" of managing form state, which is why I though it would be great to create a form hook that handles that.

Reply:
Thanks for the explanation. I definitely understand the idea of validating patterns, I just saw this one as a possibly quick fix for the EUI side to support a loading state on the button.

I 100% agree with you that EUI should be the place to look for guidance on UX behaviors. We're making guidelines one of our priorities for 2021. However, we try to be less heavy handed in the components themselves because we can't assume all use cases nor can we truly manage state from an outside library. So we've usually defaulted to allowing for customization of the UI while being state agnostic and writing guidelines for usage.